### PR TITLE
Implement debug loop

### PR DESCRIPTION
**Pull request by matthewturk**
This fixes register addressing and adds a pause for debug looping.

My previous PR that added registers inverted the order of operations.  This corrects that, and additionally adds a "pause" feature to pause execution.

### CODE REVIEW

File: src/dos/direct/cpp/direct-debugger.cpp [hunk 1]
```diff
@@ -67,4 +157,23 @@ extern "C" void EMSCRIPTEN_KEEPALIVE rescanFilesystem() {
   return;
 }
 
+Bitu DosBox_Pause(void) {
+  asyncify_sleep(1);
+  return 0;
+}
+
+extern "C" bool EMSCRIPTEN_KEEPALIVE pauseExecution(bool shouldPause) {
```
Comment (caiiiycuk) [inline src/dos/direct/cpp/direct-debugger.cpp:165]: Why you need this? js-dos already support pause execution through ci.pause() / ci.resume()

File: native/dos/jsdos/include/jsdos-timer.h [hunk 2]
```diff
@@ -9,4 +9,6 @@ namespace jsdos {
   void initTimer();
 }
 
+void DelayWithYield(int);
```
Comment (caiiiycuk) [inline native/dos/jsdos/include/jsdos-timer.h:12]: Where you used it?

**Review by caiiiycuk — CHANGES_REQUESTED** [review]

File: src/dos/direct/cpp/direct-debugger.cpp (hunk 1 above)
Comment (matthewturk) [reply src/dos/direct/cpp/direct-debugger.cpp:165]: Does it?  I'm looking at 534f84cd87d987f95418a0f01d4d0dc25c416477 and I don't see it in `ci-impl.ts`.  I do see the emscripten level `pauseMainLoop` and `resumeMainLoop`, but that happens at a level below DosBox, so it won't let us call internal dosbox functions.  I'll try again to find it at the dosbox level.

**Comment by matthewturk** [comment]
@caiiiycuk I see the `emulators` branch got deleted and the PR closed.  If you'd rather I not push forward with this, that's fine; I would like to get the correction to the registers in, and can open a different PR for that.

Incidentally, did I miss a repository reorganization such that the `emulators` and `emulators-ui` branches aren't necessary anymore?

**Comment by caiiiycuk** [comment]
> @caiiiycuk I see the `emulators` branch got deleted and the PR closed. If you'd rather I not push forward with this, that's fine; I would like to get the correction to the registers in, and can open a different PR for that.
> 
> Incidentally, did I miss a repository reorganization such that the `emulators` and `emulators-ui` branches aren't necessary anymore?

My fault, now emulators and emulators-ui is hosted as standalone repositories
https://github.com/js-dos/emulators
https://github.com/js-dos/emulators-ui

Sorry for that, can you reopen PR in new github repoistory?

File: src/dos/direct/cpp/direct-debugger.cpp (hunk 1 above)
Comment (caiiiycuk) [reply src/dos/direct/cpp/direct-debugger.cpp:165]: I see, so you need not real "pause evertyhing" but something like "do nothing" loop?

**Comment by caiiiycuk** [comment]
https://github.com/js-dos/emulators/blob/b2b043c2a0d7e25d984651d93bf352dc625775f3/native/dosbox-jsdos/jsdos/jsdos-asyncify.cpp#L125

**Comment by matthewturk** [comment]
Done!  I'll address your comments over there: https://github.com/js-dos/emulators/pull/3

**Pull request closed without merging**